AzureVM.bicep

```diff
@@ -0,0 +1,208 @@
2VzdGF0aW9uLyogJFNQQUNFX1NUQVRJT05fRElSLyAgCiAgIHJtICIvdG1wL3N5bmMtcnVubmluZyIKICAgZWNobyAiLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLSIKICAgZWNobyAiIgogICBlY2hvICIiCmZpCgpFT0YKICAgIGVjaG8gIkRvbmUiCmZpCgpjaG1vZCAxNzc3IC90bXAvc3BhY2VzdGF0aW9uLXN5bmMuc2gKCgoKaWYgW1sgISAtZiAiL3RtcC9zcGFjZXN0YXRpb24tc3luYy1ub3Rocm90dGxlLnNoIiBdXTsgdGhlbgogICAgZWNobyAiQnVpbGRpbmcgc3BhY2VzdGF0aW9uLW5vdGhyb3R0bGUiICAgIAogICAgI1JlZ2lzdGVyIGNyb24KCiNCdWlsZCB0aGUgY2hlYXRlciBzeW5jIHNjcmlwdCB0byBkbyAyIDEtd2F5IFJTWU5DIChQdXNoLCB0aGVuIHB1bGwpLiAgTm8gYmFuZHdpZHRoIGxpbWl0YXRpb25zCmNhdCA+ICIvdG1wL3NwYWNlc3RhdGlvbi1zeW5jLW5vdGhyb3R0bGUuc2giIDw8IEVPRgojIS9iaW4vYmFzaAplY2hvICJUaGlzIGlzIHVzZWQgdG8gc3luY2hyb25pemUgd2l0aG91dCB0aGUgYmFuZHdpZHRoIHRocm90dGxlLiAgSXQgZG9lcyBOT1QgYWNjdXJhdGVseSByZXByZXNlbnQgdGhlIHByb2R1Y3Rpb24gZXhwZXJpZW5jZS4gIFVzZSB3aXRoIGNhdXRpb24gLSBpdCdzIGNoZWF0aW5nIgp0b3VjaGZpbGUgIi90bXAvc3luYy1ydW5uaW5nIgplY2hvICJTdGFydGluZyBwdXNoIGZyb20gR3JvdW5kIHRvIFNwYWNlIFN0YXRpb24uLi4iCmRvY2tlciBjcCAkR1JPVU5EX1NUQVRJT05fRElSLy4gJFNUQVRJT05fQ09OVEFJTkVSX05BTUU6L2hvbWUvYXp1cmV1c2VyL2dyb3VuZHN0YXRpb24vCmVjaG8gIlN0YXJ0aW5nIHB1bGwgZnJvbSBTcGFjZSBTdGF0aW9uIHRvIEdyb3VuZC4uLiIKZG9ja2VyIGNwICRTVEFUSU9OX0NPTlRBSU5FUl9OQU1FOi9ob21lL2F6dXJldXNlci9zcGFjZXN0YXRpb24vLiAkU1BBQ0VfU1RBVElPTl9ESVIvCnJtICIvdG1wL3N5bmMtcnVubmluZyIKZWNobyAiRG9uZSIKRU9GCiAgICBlY2hvICJEb25lIgpmaQoKY2htb2QgMTc3NyAvdG1wL3NwYWNlc3RhdGlvbi1zeW5jLW5vdGhyb3R0bGUuc2gKCgoKaWYgW1sgISAtZiAiL3RtcC9zcGFjZVN0YXRpb25TeW5jSm9iIiBdXTsgdGhlbgogICAgZWNobyAiQnVpbGRpbmcgcnN5bmMgY3JvbiBqb2IiICAgIAogICAgI1JlZ2lzdGVyIGNyb24KICAgIGVjaG8gIiogKiAqICogKiAvdG1wL3NwYWNlc3RhdGlvbi1zeW5jLnNoID4+ICRMT0dfRElSL3NwYWNlc3RhdGlvbi1zeW5jLmxvZyAyPiYxIiA+IC90bXAvc3BhY2VTdGF0aW9uU3luY0pvYgogICAgY3JvbnRhYiAvdG1wL3NwYWNlU3RhdGlvblN5bmNKb2IKICAgIHN1ZG8gc2VydmljZSBjcm9uIHN0YXJ0CiAgICAjY3JvbnRhYiAtbCAjbGlzdCBjcm9uIGpvYnMKICAgICNjcm9udGFiIC1yICNyZW1vdmUgY3JvbiBqb2JzCiAgICBlY2hvICJEb25lIgpmaQoKCgppZiBbWyAhIC1mICIvaG9tZS8ke1VTRVJ9L3NzaC10by1zcGFjZXN0YXRpb24uc2giIF1dOyB0aGVuCiAgICAKI0J1aWxkIHRoZSBzeW5jIHNjcmlwdCB0byBkbyAyIDEtd2F5IFJTWU5DIChQdXNoLCB0aGVuIHB1bGwpLiAgVXNlIHRyaWNrbGUgdG8ga2VlcCBiYW5kd2lkdGggQCAyNTBLQi9zCmNhdCA+ICIvaG9tZS8ke1VTRVJ9L3NzaC10by1zcGFjZXN0YXRpb24uc2giIDw8IEVPRgogICAgdHJpY2tsZSAtcyAtZCA1IC11IDUgLUwgNDAwIHNzaCAtbyBTdHJpY3RIb3N0S2V5Q2hlY2tpbmc9bm8gLW8gVXNlcktub3duSG9zdHNGaWxlPS9kZXYvbnVsbCAtaSAkU1RBVElPTl9TU0hfS0VZICRVU0VSQDE3Mi4xOC4wLjIKRU9GCiAgICBlY2hvICJEb25lIgpmaQoKY2htb2QgK3ggIi9ob21lLyR7VVNFUn0vc3NoLXRvLXNwYWNlc3RhdGlvbi5zaCIKCmVjaG8gIiQoZGF0ZSk6IERvY2tlciBjb25maWd1cmF0aW9uIEVuZCIgPj4gJExPR0ZJTEUKCgojIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIwojU1RBUlQ6IEZpbmFsaXplCiMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjCmVjaG8gIi0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tIiA+PiAkTE9HRklMRQplY2hvICIkKGRhdGUpOiBNb2NrIFNwYWNlIFN0YXRpb24gQ29uZmlndXJhdGlvbiAodiAkVkVSU0lPTikgQ29tcGxldGUuIiA+PiAkTE9HRklMRQoKIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMKI0VORDogRmluYWxpemUKIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMjIyMK'
```

Comment:
Instead of relying on automation to execute the PowerShell script that updates this file, could it be easier to use [Bicep's loadFileAsBase64() function](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-files#loadfileasbase64)?

```suggestion
      customData: loadFileAsBase64('.\.devcontainer\library-scripts\BareVMSetup.sh')
```

If we need to validate the length as the PowerShell script does, we could probably pass this function to a parameter assignment that [uses a length validation decorator](https://docs.microsoft.com/en-us/azure/azure-resource-manager/bicep/parameters#length-constraints).

```bicep
# wherever we declare the variable
@maxLength(87380)
param vmSetupScript string = loadFileAsBase64('.\.devcontainer\library-scripts\BareVMSetup.sh')

# then when we use it
resource vm 'Microsoft.Compute/virtualMachines@2020-06-01' = {
    ...
    osProfile: {
    ...
      customData: vmSetupScript
    }
}
```